### ISSUE TITLE
Problem starting MSWord 2007 or 2013 under XP using NVDA 2016.2

### ISSUE THREAD

**Issue by MichelSuch** [nvaccess/nvda#6033]
Hi,

I know that Windows XP is nomore really supported by NVDA, but many people still use it.
To-day, I have been reported that Word 2007 and 2013 refuse to start with NVDA 2016.2.
This did not happen with previous versions.
Is there something that could be done to fix this?

**Comment by derekriemer** [comment]
Actually, NVDA still supports XP.
Can you give more info (like does an error occur)? Also, are you sure 
that word 2013 is supported on XP?
On 6/2/2016 10:41 AM, MichelSuch wrote:
>
> Hi,
>
> I know that Windows XP is nomore really supported by NVDA, but many 
> people still use it.
> To-day, I have been reported that Word 2007 and 2013 refuse to start 
> with NVDA 2016.2.
> This did not happen with previous versions.
> Is there something that could be done to fix this?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nvaccess/nvda/issues/6033>, or mute the thread 
> <https://github.com/notifications/unsubscribe/AFGivSoKKoFpd7TYHAO6_FPSMKiWj8Zeks5qHwfXgaJpZM4IswKH>.
>

-- 
------------------------------------------------------------------------


    Derek Riemer

  * Department of computer science, third year undergraduate student.
  * Proud user of the NVDA screen reader.
  * Open source enthusiast.
  * Member of Bridge Cu
  * Avid skiier.

Websites:
Honors portfolio <http://derekriemer.com>
Awesome little hand built weather app! 
<http://django.derekriemer.com/weather/>

email me at derek.riemer@colorado.edu <mailto:derek.riemer@colorado.edu>
Phone: (303) 906-2194

**Comment by MichelSuch** [comment]
Hi Derek,


Sorry, I mansionned Word 2013, this was wrong, the problem has only been 
reported for 2007 version.

At this point, I understand that NVDA prevents Word from starting, but 
I've asked for more details because I find this strange.

I'll let you know when I get more details.


                     Michel SUCH

michel.such@free.fr
Skype : michel.such
Twitter : @SuchMichel

**Comment by josephsl** [comment]
Hi,

Which XP version is this – RTM, SP1, SP2, SP3? NVDA 2016.2 will require XP SP2 or later, and the version of Office that’ll work fine would be 2007 or 2010 (Office 2013 requires certain CPU features).

Cheers,

Joseph

**Comment by MichelSuch** [comment]
Hi Joseph,


XP SP3.


                     Michel SUCH

michel.such@free.fr
Skype : michel.such
Twitter : @SuchMichel

**Comment by josephsl** [comment]
Hi,

Can you tell your friend: when the error happens again, can you attach both nvda.log and nvdaOld.log to this ticket so one of us can examine it? Thanks.

Cheers,

Joseph

**Comment by derekriemer** [comment]
Remember to remove any sensitive information from his log.


On 6/2/2016 11:59 AM, josephsl wrote:
> Hi,
>
> Can you tell your friend: when the error happens again, can you attach 
> both nvda.log and nvdaOld.log to this ticket so one of us can examine 
> it? Thanks.
>
> Cheers,
>
> Joseph
>
>
>
> From: MichelSuch [mailto:notifications@github.com]
> Sent: Thursday, June 2, 2016 10:50 AM
> To: nvaccess/nvda <nvda@noreply.github.com>
> Cc: josephsl <joseph.lee22590@gmail.com>; Comment 
> <comment@noreply.github.com>
> Subject: Re: [nvaccess/nvda] Problem starting MSWord 2007 or 2013 
> under XP using NVDA 2016.2 (#6033)
>
>
>
> Hi Joseph,
>
>
> XP SP3.
>
>
> Michel SUCH
>
> michel.such@free.fr
> Skype : michel.such
> Twitter : @SuchMichel
>
> Le 2016-06-02 à 19:48, josephsl a écrit :
> > Hi,
> >
> > Which XP version is this – RTM, SP1, SP2, SP3? NVDA 2016.2 will
> > require XP SP2 or later, and the version of Office that’ll work fine
> > would be 2007 or 2010 (Office 2013 requires certain CPU features).
> >
> > Cheers,
> >
> > Joseph
> >
> >
> >
> > From: MichelSuch [mailto:notifications@github.com]
> > Sent: Thursday, June 2, 2016 9:42 AM
> > To: nvaccess/nvda <nvda@noreply.github.com>
> > Subject: [nvaccess/nvda] Problem starting MSWord 2007 or 2013 under XP
> > using NVDA 2016.2 (#6033)
> >
> >
> >
> > Hi,
> >
> > I know that Windows XP is nomore really supported by NVDA, but many
> > people still use it.
> > To-day, I have been reported that Word 2007 and 2013 refuse to start
> > with NVDA 2016.2.
> > This did not happen with previous versions.
> > Is there something that could be done to fix this?
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/nvaccess/nvda/issues/6033> , or mute the thread
> > 
> <https://github.com/notifications/unsubscribe/AHgLkKsxP4o-ryRPKPwQ-ahjTeoS18nSks5qHwfXgaJpZM4IswKH> 
>
> > .
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/nvaccess/nvda/issues/6033#issuecomment-223368409>,
> > or mute the thread
> > 
> <https://github.com/notifications/unsubscribe/APFCjxeg8eCT4U2xb-QgTyVsrjOXnAF5ks5qHxeDgaJpZM4IswKH>.
> >
>
>
>
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nvaccess/nvda/issues/6033#issuecomment-223368691> 
> , or mute the thread 
> <https://github.com/notifications/unsubscribe/AHgLkEfU8RB8TAVUNjj5Mau7F_20aLm3ks5qHxe9gaJpZM4IswKH> 
> .
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nvaccess/nvda/issues/6033#issuecomment-223371576>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AFGivUyoX0L82_R4BnlDcFwURMMfdkmhks5qHxn5gaJpZM4IswKH>.
>

-- 
------------------------------------------------------------------------


    Derek Riemer

  * Department of computer science, third year undergraduate student.
  * Proud user of the NVDA screen reader.
  * Open source enthusiast.
  * Member of Bridge Cu
  * Avid skiier.

Websites:
Honors portfolio <http://derekriemer.com>
Awesome little hand built weather app! 
<http://django.derekriemer.com/weather/>

email me at derek.riemer@colorado.edu <mailto:derek.riemer@colorado.edu>
Phone: (303) 906-2194

**Comment by MichelSuch** [comment]
Hi,


I've got soe more information.

In fact, NVDA does not crash.

Word starts, and then closes at once.

I cannot imagine how this is possible, but reported by 2 users.


                     Michel SUCH

michel.such@free.fr
Skype : michel.such
Twitter : @SuchMichel

**Comment by Brian1Gaff** [comment]
First thing to check is to unload nvda and see if the same thing occurs. Its 
always possible that an office update has screwed the programs up when you 
run them in XP.
 I have the hack on XP here to get system updates, so cannot try this idea 
out easily.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
----- Original Message -----

**Comment by MichelSuch** [comment]
Hi Brian,


No, they told me that when NVDA is not loaded, Word starts normally.


                     Michel SUCH

michel.such@free.fr
Skype : michel.such
Twitter : @SuchMichel

**Comment by josephsl** [comment]
CC @MichaelDCurran

**Comment by Brian1Gaff** [comment]
Right, I've found an issue sometimes when word of any vintage starts, the 
main text field does not work. Sometimes alttab and back restores it at 
others I get Microsoft word has stopped responding, and a file with a $ at 
the start in the current folder I'm using
This happens in any operating system but not all the time. It can occur in 
both loading  it via loading the document, or just a blank new occurence.
 Its as if something is running aat the worng time sometimes and screwing up 
Word.
 I'm not sure how far back this might go, as I might have blamed the file or 
Windows before, but since you say no nvda and it works, that is odd.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
----- Original Message -----

**Comment by jcsteh** [comment]
Confirmed. This particular issue seems to be specific to XP and is 100% reproduceable. (We believe intermittent Word crashes with other Windows versions are unrelated.)

Technical: Here's the crash stack.

Note that i forgot to load symbols for this first bit, so ignore the symbol name:

```
(1d20.1d14): Access violation - code c0000005 (first chance)
First chance exceptions are reported before any exception handling.
This exception may be expected and handled.
eax=00000000 ebx=00022280 ecx=02eae420 edx=00000000 esi=02eae420 edi=02eae450
eip=10017b79 esp=0012f448 ebp=0012f498 iopl=0         nv up ei ng nz na po nc
cs=001b  ss=0023  ds=0023  es=0023  fs=003b  gs=0000             efl=00010282
*** ERROR: Symbol file could not be found.  Defaulted to export symbols for C:\Program Files\NVDA\lib\nvdaHelperRemote.dll - 
nvdaHelperRemote!nvdaControllerInternal_installAddonPackageFromPath+0x1419:
10017b79 8b1490          mov     edx,dword ptr [eax+edx*4] ds:0023:00000000=????????
```

Symbols are loaded correctly from here.

```
0:000> kp
ChildEBP RetAddr  
0012f498 10017dec nvdaHelperRemote!getHeadingLevelFromParagraph(struct IDispatch * pDispatchParagraph = 0x02eae420)+0x39 [c:\projects\nvda\build\x86\remote\winword.cpp @ 367]
0012f4ac 1001998c nvdaHelperRemote!generateHeadingXML(struct IDispatch * pDispatchParagraph = 0x02eae420, struct IDispatch * pDispatchParagraphRange = 0x02eae450, int startOffset = 0n0, int endOffset = 0n1, class std::basic_ostringstream<wchar_t,std::char_traits<wchar_t>,std::allocator<wchar_t> > * XMLStream = 0x0012f568)+0xc [c:\projects\nvda\build\x86\remote\winword.cpp @ 412]
0012f6c8 1001ac8a nvdaHelperRemote!winword_getTextInRange_helper(struct HWND__ * hwnd = <Value unavailable error>, struct winword_getTextInRange_args * args = 0x00000000, struct std::pair<long,long> * _Parg = <Value unavailable error>, unsigned int _Count = <Value unavailable error>, void * _Ptr = <Value unavailable error>)+0x52c [c:\projects\nvda\build\x86\remote\winword.cpp @ 878]
0012f6d0 10009f98 nvdaHelperRemote!winword_callWndProcHook(int code = 0n0, unsigned int wParam = 1, long lParam = 0n1242948)+0x3a [c:\projects\nvda\build\x86\remote\winword.cpp @ 1055]
0012f710 7e42b372 nvdaHelperRemote!inProcess_callWndProcHook(int code = 0n0, unsigned int wParam = 1, long lParam = 0n1242948, void * <_Args_0> = <Value unavailable error>, struct std::_Tree_node<std::pair<long (__stdcall*const)(int,unsigned int,long),unsigned int>,void *> * _Pnode = 0x001c6b00)+0xe8 [c:\projects\nvda\build\x86\remote\inprocess.cpp @ 142]
0012f72c 7e4565b7 USER32!DispatchHookW+0x31
0012f754 7e428eec USER32!fnHkINLPCWPSTRUCTW+0x4f
0012f77c 7c90e453 USER32!__fnDWORD+0x24
0012f7a0 7e4193e9 ntdll!KiUserCallbackDispatcher+0x13
0012f7cc 7e419402 USER32!NtUserPeekMessage+0xc
*** ERROR: Symbol file could not be found.  Defaulted to export symbols for C:\PROGRA~1\MICROS~2\Office12\wwlib.dll - 
0012f7f8 312a0dbb USER32!PeekMessageW+0xbc
WARNING: Stack unwind information not available. Following frames may be wrong.
0012f814 312a0d73 wwlib!DllGetClassObject+0x5c171
0012f838 312a0d36 wwlib!DllGetClassObject+0x5c129
0012f87c 312a3e9b wwlib!DllGetClassObject+0x5c0ec
0012f8b0 312a3a3c wwlib!DllGetClassObject+0x5f251
0012f8c4 312a38ee wwlib!DllGetClassObject+0x5edf2
0012f944 312bd7c0 wwlib!DllGetClassObject+0x5eca4
0012f958 312a27ab wwlib!DllGetClassObject+0x78b76
0012f980 31244c0e wwlib!DllGetClassObject+0x5db61
*** ERROR: Symbol file could not be found.  Defaulted to export symbols for C:\PROGRA~1\MICROS~2\Office12\WINWORD.EXE - 
0012ff0c 300015d7 wwlib!FMain+0x6ac
0012ff30 3000155d WINWORD+0x15d7
0012ffc0 7c817067 WINWORD+0x155d
0012fff0 00000000 kernel32!BaseProcessStart+0x23
```

**Comment by michaelDCurran** [comment]
Microsoft Connect bug: https://connect.microsoft.com/VisualStudio/Feedback/Details/1941836

**Comment by nvaccessAuto** [comment]
Incubated in 8b4b002adb294960014ba6b70cde659b8830c4eb.

**Comment by MichelSuch** [comment]
Hi,


User tells me that it works fine with latest next.

So, for me, this ticket can be closed.

Will there be a 2016.2.1 release to fix this, or should I tell to user 
to come on with this next release?


                     Michel SUCH

michel.such@free.fr
Skype : michel.such
Twitter : @SuchMichel

**Comment by jcsteh** [comment]
Fix included in NVDA 2016.2.1, which was released a few hours ago.
